### ISSUE TITLE
enforcedHostRegexp does not work with empty string, works with one blank space

### ISSUE THREAD

**Issue by sitilge** [linkerd/linkerd2#6969]
### What is the issue?

Using an empty string in `dashboard.enforcedHostRegexp=""` (as explained in docs) is still producing:

```
It appears that you are trying to reach this service with a host of 'linkerd.abc.xyz'.
This does not match /^(localhost|127\.0\.0\.1|web\.linkerd-viz\.svc\.cluster\.local|web\.linkerd-viz\.svc|\[::1\])(:\d+)?$/ and has been denied for security reasons.
Please see https://linkerd.io/dns-rebinding for an explanation of what is happening and how to fix it.
```

### Solution

Using a string with one blank space solves it:

```
        dashboard:
          enforcedHostRegexp: " "
```

**Comment by alpeb** [comment]
Thanks for the report :+1: 
I think the best way to address this is by recommending to use a catch-all regexp instead of the empty string (because Helm doesn't allow us to distinguish between this value being empty and it not being set). I've issued linkerd/website#1194 to amend the docs.

**Comment by adleong** [comment]
Closed in favor of linkerd/website#1194
